targets/TARGET_NXP/TARGET_MCUXpresso_MCUS/TARGET_LPC54114/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+if("LPCXpresso" IN_LIST MBED_TARGET_LABELS)
+    target_include_directories(mbed-core
+        INTERFACE
+            TARGET_LPCXpresso
+    )
+
+    target_sources(mbed-core
+        INTERFACE
+            TARGET_LPCXpresso/PeripheralPins.c
+            TARGET_LPCXpresso/clock_config.c
+            TARGET_LPCXpresso/mbed_overrides.c
+    )
+endif()
+
+if("LPC54114_M4" IN_LIST MBED_TARGET_LABELS)
+    if(${MBED_TOOLCHAIN} STREQUAL "ARM")
+        set(STARTUP_FILE device/TARGET_LPC54114_M4/TOOLCHAIN_ARM_STD/startup_LPC54114_cm4.S)
+        set(LINKER_FILE device/TARGET_LPC54114_M4/TOOLCHAIN_ARM_STD/LPC54114J256_cm4.sct)
+        set(LIB_POWER device/TARGET_LPC54114_M4/TOOLCHAIN_ARM_STD/libpower.ar)
+    elseif(${MBED_TOOLCHAIN} STREQUAL "GCC_ARM")
+        set(STARTUP_FILE device/TARGET_LPC54114_M4/TOOLCHAIN_GCC_ARM/startup_LPC54114_cm4.S)
+        set(LINKER_FILE device/TARGET_LPC54114_M4/TOOLCHAIN_GCC_ARM/LPC54114J256_cm4_flash.ld)
+        set(LIB_POWER device/TARGET_LPC54114_M4/TOOLCHAIN_GCC_ARM/libpower.a)
+    endif()
+
+    target_link_libraries(mbed-core INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/${LIB_POWER})
+
+    target_include_directories(mbed-core
+        INTERFACE
+            device/TARGET_LPC54114_M4
+    )
+
+    target_sources(mbed-core
+        INTERFACE
+            device/TARGET_LPC54114_M4/system_LPC54114_cm4.c
+    )
+endif()
+
+set_property(
```

Comment:
can we have this on the same line (length is similar as lines above). Somewhere below 100 char

Reply:
It was not done because of line length it was done to help readability. I do not know if I have indented it as usually done by the CMake but I tried 😄

Reply:
[Done](https://github.com/ARMmbed/mbed-os/compare/4d28deb659053b6002bff1ec37b9019a4ce24174..e8197405022dee475094cadb030f8e511a1fec40).

Reply:
We have these as one liners, so either we refactor all and follow or stay consistent and make this one liner as well. I vote for one liner for set() and similar. It could be split if the length is too long, but that should not be the case as then it would indicate the path is long and we have probably too few components.